src/completion.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const stdx = @import("stdx.zig");
+const assert = std.debug.assert;
+
+const RingBuffer = @import("ring_buffer.zig").RingBufferType;
+const BoundedArray = stdx.BoundedArrayType;
+
+const completion_entries = 256;
+const completion_entry_bytes = 512;
+const keywords = [_][]const u8{
+    "create_accounts",
+    "create_transfers",
+    "lookup_accounts",
+    "lookup_transfers",
+    "get_account_transfers",
+    "get_account_balances",
+    "query_accounts",
+    "query_transfers",
+    "help",
+    "id",
+    "code",
+    "ledger",
+    "flags",
+    "account_id",
+    "debit_account_id",
+    "credit_account_id",
+    "amount",
+};
+
+pub const Completion = struct {
+    const CompletionList = RingBuffer([completion_entries]u8, .slice);
+
+    matches: CompletionList,
+    prefix: BoundedArray(u8, completion_entry_bytes),
+    suffix: BoundedArray(u8, completion_entry_bytes),
+    query: BoundedArray(u8, completion_entry_bytes),
+    match_index: usize = 0,
+
+    pub fn init(self: *Completion, allocator: std.mem.Allocator) !void {
+        self.* = Completion{
+            .matches = try CompletionList.init(allocator, completion_entry_bytes),
```

Comment:
As mentioned above, I think this constant should be `completion_entries`, not `completion_entry_bytes`.